internal/ui/esbuild.go

```diff
@@ -0,0 +1,161 @@
+var buildOptions = api.BuildOptions{
+	Outdir:      fmt.Sprintf("%s/dist", env.PublicDir()),
+	Bundle:      true,
+	Write:       true,
+	TreeShaking: api.TreeShakingTrue,
+	EntryNames:  "[name]-[hash]",
+	Splitting:   true,
+	Format:      api.FormatESModule,
+	Loader: map[string]api.Loader{
+		".scss": api.LoaderCSS, // Treat SCSS files as CSS
+	},
+	Define: map[string]string{
+		"process.env.NODE_ENV": "\"development\"", // Required to define mode when minifying files, or it will default to production
+	},
+	Plugins: []api.Plugin{
+		{
+			Name: "HTML Plugin",
+			Setup: func(setup api.PluginBuild) {
+				setup.OnEnd(func(result *api.BuildResult) (api.OnEndResult, error) {
+					if err := createHtml(result.OutputFiles); err != nil {
+						log.Printf("failed to create index.html: %v", err)
+					}
+					return api.OnEndResult{}, nil
+				})
+			},
+		},
+	},
+}
+
+// getOptions returns the build options for esbuild
+func getOptions(outputDir *string, dev bool) api.BuildOptions {
+	// Dynamically add entry points to the build options
+	// Adding more can reduce the size of the output files, but will increase the file count
+	entryPoints := []string{
+		"src/index.tsx",
+		"src/App.tsx",
+		"src/overmind/state.tsx",
+		"src/overmind/actions.tsx",
+		"src/overmind/effects.tsx",
+		"src/overmind/index.tsx",
+		"src/components/Results.tsx",
+	}
+	var entries []string
+	for _, entry := range entryPoints {
+		entries = append(entries, fmt.Sprintf("%s/%s", env.PublicDir(), entry))
+	}
+	buildOptions.EntryPoints = entries
+	buildOptions.LogLevel = api.LogLevelDebug
+	buildOptions.Sourcemap = api.SourceMapLinked
+	buildOptions.MinifyWhitespace = true
+	buildOptions.MinifyIdentifiers = true
+	buildOptions.MinifySyntax = true
+
+	if !dev {
+		buildOptions.Define["process.env.NODE_ENV"] = "production"
+		buildOptions.LogLevel = api.LogLevelError
+	}
+
+	// This is done to enable testing without overwriting current build
+	if outputDir != nil {
+		buildOptions.Outdir = *outputDir
+	}
+	return buildOptions
+}
+
+// resetDistFolder removes the dist folder and creates a new one
+func resetDistFolder() error {
+	path := fmt.Sprintf("%s/dist", env.PublicDir())
+	if _, err := os.Stat(path); err == nil {
+		if err := os.RemoveAll(path); err != nil {
+			return fmt.Errorf("failed to remove dist directory: %v", err)
+		}
+	}
+	if err := os.MkdirAll(path, 0o755); err != nil {
+		return fmt.Errorf("failed to create dist directory: %v", err)
+	}
+	return nil
+}
+
+// Build builds the UI with esbuild
+// The output is public/dist
+// Scss files are treated as css
+func Build(outputDir *string, dev bool) error {
+	if err := resetDistFolder(); err != nil {
+		return fmt.Errorf("failed to reset dist folder: %v", err)
+	}
+	result := api.Build(getOptions(outputDir, dev))
+	if len(result.Errors) > 0 {
+		return fmt.Errorf("failed to build UI: %v", parseMessages(result.Errors))
+	}
+	if err := createHtml(result.OutputFiles); err != nil {
```

Comment:
With the plugin HTML plugin this is no longer necessary here. Otherwise it will run twice, once here and once by the plugin.